README.md

```diff
@@ -43,87 +43,33 @@ ICICLE is a CUDA implementation of general functions widely used in ZKP. ICICLE
+Checkout [Google Colab](https://colab.google/). Google Colab offers a free [T4 GPU](https://www.nvidia.com/en-us/data-center/tesla-t4/) instance and ICICLE can be used with it, refrence this guide for setting up your [Google Colab workplace][GOOGLE-COLAB-ICICLE].
 
-```sh
-mkdir -p build
-nvcc -o build/<binary_name> ./icicle/curves/index.cu -lib -arch=native
-```
-
-### Testing the CUDA code
-
-We are using [googletest] library for testing. To build and run [the test suite](./icicle/README.md) for finite field and elliptic curve arithmetic, run from the `icicle` folder:
-
-For testing, ensure the `BUILD_TESTS` option is enabled in cmake. If not, toggle it on by adding `-DBUILD_TESTS=ON` in the cmake configuration command:
-
-```sh
-cmake -S . -B build -DBUILD_TESTS=ON
-```
+If you require more compute and have an interesting research project, we have a [bounties and grants program][GRANT_PROGRAM].
 
-Proceed with the following commands:
-
-```sh
-mkdir -p build
-cmake -S . -B build
-cmake --build build
-cd build && ctest
-```
-
-NOTE: If you are using cmake versions < 3.24 add `-DCUDA_ARCH=<target_cumpute_arch>` to the command `cmake -S . -B build`
 
+### Build systems
 
-### Rust Bindings
+ICICLE has three build systems.
 
-For convenience, we also provide rust bindings to the ICICLE library for the following primitives:
+- [ICICLE core][ICICLE-CORE], C++ and CUDA
+- [ICICLE Rust][ICICLE-RUST] bindings
+- [ICICLE Golang][ICICLE-GO] bindings
 
-- MSM
-- NTT
-    - Forward NTT
-    - Inverse NTT
-- ECNTT
-    - Forward ECNTT
-    - Inverse NTT
-- Scalar Vector Multiplication
-- Point Vector Multiplication
+ICICLE core always needs to be build as part of the other build systems as it containse the core ICICLE primitves implemeted in Cuda. Refrence these guides for the different build systems, [ICICLE core guide][ICICLE-CORE-README], [ICICLE Rust guide][ICICLE-RUST-README] and [ICICLE Golang guide][ICICLE-GO-README].
 
-A custom [build script][B_SCRIPT] is used to compile and link the ICICLE library. The environment variable `ARCH_TYPE` is used to determine which GPU type the library should be compiled for and it defaults to `native` when it is not set allowing the compiler to detect the installed GPU type.
-
-> NOTE: A GPU must be detectable and therefore installed if the `ARCH_TYPE` is not set.
-
-Once you have your parameters set, run:
-
-```sh
-cargo build --release
-```
-
-You'll find a release ready library at `target/release/libicicle_utils.rlib`.
-
-To benchmark and test the functionality available in RUST, run:
-
-```
-cargo bench
-cargo test -- --test-threads=1
-```
-
-The flag `--test-threads=1` is needed because currently some tests might interfere with one another inside the GPU.
-
-### Example Usage
-
-An example of using the Rust bindings library can be found in our [fast-danksharding implementation][FDI]
-
-### Supporting Additional Curves
+# Supporting Additional Curves
 
 Supporting additional curves can be done as follows:
```

Comment:
Honestly, I don't think it was ever possible to support most curves by just running our script, but now it's completely broken. I would remove curve adding guide tbh